types.cc

```diff
@@ -752,7 +752,7 @@ struct is_byte_order_equal_visitor {
     bool operator()(const reversed_type_impl& t) { return t.underlying_type()->is_byte_order_equal(); }
     bool operator()(const string_type_impl&) { return true; }
     bool operator()(const bytes_type_impl&) { return true; }
-    bool operator()(const date_type_impl&) { return true; }
+    bool operator()(const concrete_type<db_clock::time_point>&) { return true; }
```

Comment:
At the very least we need an alias for this, with the old name and new implementation.

Reply:
Not sure I understand. The base class for date_type_impl and timestamp_type_impl is concrete_type<db_clock::time_point>. That is how this patch removes duplicated code.

I can add a

using timestamp_date_base_class = concrete_type<db_clock::time_point>;

Is that what you are suggesting?